### PR TITLE
Display FOV Darken/Highlight and FieldOfFire in more phases

### PR DESCRIPTION
**Pull request by pakfront**
This fixes https://github.com/MegaMek/megamek/issues/4327 and fixes https://github.com/MegaMek/megamek/issues/4400. It shows the optional line-of-sight Darken/Highlight  in all non-report phases, rather than just Movement. It shows the optional weapon Field of Fire during the deployment phase.
 Reviewers, please let me know if this new feature should be behind a client option, or if the highlighting should be enabled during reports as well.

### CODE REVIEW

**Comment by kuronekochomusuke** [comment]
would probably be better as a separate PR to get it working in the reporting phases.  since it currently only works on the "selected unit" and would need to have it work on any unit without it being confusing.

File: megamek/src/megamek/client/ui/swing/FiringDisplay.java [hunk 1]
```diff
@@ -2676,17 +2676,17 @@ private boolean validStrafingCoord(Coords newCoord) {
         return isConsecutive && isInaLine;
     }
 
-    public void FieldofFire(Entity unit, int[][] ranges, int arc, int loc, int facing) {
+    public void setWeaponFieldofFire(Entity unit, int[][] ranges, int arc, int loc, int facing) {
```
[the remark this answers is not in the document]
Comment (pakfront) [reply megamek/src/megamek/client/ui/swing/FiringDisplay.java:2679]: Good call, will do.

[the remark this answers is not in the document]
Comment (pakfront) [reply megamek/src/megamek/client/ui/swing/FiringDisplay.java]: I moved it up to StatusBarDisplay.

**Review by SJuliez — APPROVED** [review]

**Comment by SJuliez** [comment]
This has picked up a merge conflict. Can you please resolve that, then I'll merge it; thanks!

**Pull request merged**